src/main/proto/ga4gh/read_service.proto

```diff
@@ -91,6 +99,52 @@ message GetReadGroupSetRequest {
   string read_group_set_id = 1;
 }
 
+// ******************  /readgroups  *********************/
+
+// This request maps to the body of `POST /readgroups/search` as JSON.
+
+
+message SearchReadGroupsRequest {
+  // The dataset to search.
+  string dataset_id = 1;
+  
+  // Only return readgroups within the provided read group set ID.
+  string read_group_set_id = 2;
+
+  // Only return read groups with this name (case-sensitive, exact match).
+  string name = 3;
+
+  //  Specifying the id of a BioSample record will return only readgroups
+  //  with the given bioSampleId.
+  string bio_sample_id = 4;
```

Comment:
Within a single read group set each read group can have its own `bio_sample_id`. This is to model BAMs that may have different samples per RG tag.